openshift-knative-operator/pkg/monitoring/monitoring.go

```diff
@@ -0,0 +1,93 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	mf "github.com/manifestival/manifestival"
+	"github.com/openshift-knative/serverless-operator/openshift-knative-operator/pkg/common"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
+)
+
+const (
+	DisableMonitoringEnvVar      = "DISABLE_SERVING_MONITORING_BY_DEFAULT"
+	EnableMonitoringLabel        = "openshift.io/cluster-monitoring"
+	ObservabilityCMName          = "observability"
+	ObservabilityBackendKey      = "metrics.backend-destination"
+	OpenshiftMonitoringNamespace = "openshift-monitoring"
+)
+
+func SetupServingMonitoring(api kubernetes.Interface, ks *v1alpha1.KnativeServing, log *zap.SugaredLogger) error {
+	if shouldDisableMonitoring(ks.Spec.CommonSpec.Config) {
+		log.Info("Disabling Serving monitoring")
+		if err := disableMonitoring(api, ks); err != nil {
+			return fmt.Errorf("failed to disable monitoring %w ", err)
+		}
+		return nil
+	}
+	if err := setMonitoringLabelToNamespace(ks.Namespace, api, true); err != nil {
+		return err
+	}
+	return nil
+}
+
+func shouldDisableMonitoring(cfg v1alpha1.ConfigMapData) bool {
+	_, disableByDefault := os.LookupEnv(DisableMonitoringEnvVar)
```

Comment:
What if I set the env var to `false`? Should this parse the env var to an extend?

Reply:
This does not check the value but it could

Reply:
just my 2cents: Typically it's better to formulate flags in a 'positive' way, to avoid double negative (like DISABLE_...=false). IMO its a better user experience to use `ENABLE_SERVICE_MONITORING_BY_DEFAULT` (or maybe even more explicit `ENABLE_SERVICE_MONITORING_AS_FALLBACK`) with a value of `false` for this behaviour.

This would be especially important if there are already ENABLE_ kind of environment variables (don't know if this is the case here, but the usage should be consistent, and introducing DISABLE_ now would imply that all future boolean env vars should use DISABLE_ , too)

I don't insist on a change.